source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+
+	regenerateFilterSummary();
+}
+
+function createDetailsDialog(timelineFilter: Element): void {
+	const detailsDialog = select('details-dialog', timelineFilter)!;
+
+	detailsDialog.setAttribute('src', '');
+
+	detailsDialog.setAttribute('aria-label', 'Timeline filter settings');
+	select('div.Box-header h3', detailsDialog)!.textContent = 'Timeline filter settings';
+
+	// Close button should restore previous settings.
+	select('div.Box-header button', detailsDialog)!.addEventListener('click', restoreSettings);
+
+	const form = <div/>;
+
+	createItem(form, hideResolvedSelectorId, 'Hide resolved comments', '', CurrentSettings.HideResolved, false);
+	createItem(form, hideCommitsSelectorId, 'Hide commits', '', CurrentSettings.HideCommits, true);
+	createItem(form, hideUnresolvedSelectorId, 'Hide unresolved comments', '', CurrentSettings.HideUnresolved, true);
+	createItem(form, hideNormalCommentsSelectorId, 'Hide normal comments', 'Hides comments that does not contain unresolved/resolved state.', CurrentSettings.hideNormalComment, true);
+	createItem(form, hideOthersSelectorId, 'Hide other', 'Hides any other kind of activity that was not specified above.', CurrentSettings.HideOthers, true);
+	createItem(form, autoLoadHiddenSelectorId, 'Load hidden', 'Automatically loads hidden timeline items.', CurrentSettings.AutoLoadHidden, true);
+
+	const actionButtons = (
+		<div className="Box-footer form-actions">
+			<button type="submit" className="btn btn-primary" data-disable-with="Saving…" onClick={async () => saveSettings()}>Save</button>
+			<button type="reset" className="btn" data-close-dialog="" onClick={() => restoreSettings()}>Cancel</button>
+		</div>
+	);
+
+	form.append(actionButtons);
+
+	// This works on github enterprise - form is already preloaded
+	select('form', detailsDialog)?.remove();
+	// This works on normal github. Normally form is loaded in place of `include-fragment` after we open details dialog.
+	select('include-fragment', detailsDialog)?.remove();
+
+	detailsDialog.append(form);
+}
+
+async function tryClickLoadMore(item: HTMLElement): Promise<any> {
+	if (CurrentSettings.AutoLoadHidden) {
+		// Just after loading page when user clicks that element he is redirected to some limbo. It happens because github javascript did not kick in yet.
+		// To mitigate that we always give 1 second for javascript to load and notice this element so clicking it will be handled properly.
+		await sleep(1);
+		item.click();
+	}
+}
+
+function processTimelineItem(item: HTMLElement): void {
+	const pr = select('.js-comment[id^=pullrequestreview]', item);
+	const commitGroup = select('.js-commit-group', item);
+	const normalComment = select('.js-comment-container', item);
+
+	if (pr) {
+		processPR(item);
+	} else if (commitGroup) {
+		applyDisplay(item, CurrentSettings.HideCommits);
+	} else if (normalComment) {
+		applyDisplay(item, CurrentSettings.hideNormalComment);
+	} else {
+		applyDisplay(item, CurrentSettings.HideOthers);
```

Comment:
What do you think about this:

- process each item once, only to add a class (e.g. `rgh-is-review-comment`)
- when the settings change, only change a single class on the timeline container (`.rgh-hide-reviews .rgh-is-review-comment {display: none}`
- avoid processing items at all if they’re already targetable without the custom class: `.rgh-hide-reviews .js-comment[id^=pullrequestreview] {display: none}`

Reply:
This change can come later, let’s fix the UI and rest of the code first.